ui/selectors/selectors.js

```diff
@@ -1137,47 +1137,48 @@ export function getPetnamesEnabled(state) {
   return petnamesEnabled;
 }
 
-export function getIsTokenNetworkFilterEqualCurrentNetwork(state) {
-  const chainId = getCurrentChainId(state);
-  const tokenNetworkFilter = getTokenNetworkFilter(state);
-  if (
-    Object.keys(tokenNetworkFilter).length === 1 &&
-    Object.keys(tokenNetworkFilter)[0] === chainId
-  ) {
-    return true;
-  }
-  return false;
-}
-
 /**
  * Returns an object indicating which networks
  * tokens should be shown on in the portfolio view.
  *
  * @param {*} state
  * @returns {Record<Hex, boolean>}
  */
-export function getTokenNetworkFilter(state) {
-  const currentChainId = getCurrentChainId(state);
-  const { tokenNetworkFilter } = getPreferences(state);
+// THIS KEPT RE-RENDERING
+export const getTokenNetworkFilter = createDeepEqualSelector(
```

Comment:
This I think is the biggest culprit. It kept re-rendering the TokenList component.